api/src/main/java/io/minio/CopySource.java

```diff
@@ -0,0 +1,97 @@
+package io.minio;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/** SourceInfo creates the SourceObject
+ * used for Compose Object.
+ */
+public class CopySource {
+  private String bucketName;
+  private String objectName;
+  private int offset;
+  private int length;
```

Comment:
You would need to use `Integer` than `int` so that they can be set to `null`.

* `offset = null && length == null`, return `x-amz-copy-source-range` header is not set.
* `offset != null && length != null`, return header `x-amz-copy-source-range=byte=<offset>-<offset+length>`
* `offset != null && length == null`, return header `x-amz-copy-source-range=byte=<offset>-`
* `offset == null && length != null`, return header `x-amz-copy-source-range=byte=0-<length>`

Reply:
done.